File: java/Figuras/src/main/java/org/example/Cubo.java
package org.example;

/**
 * Clase Cubo - Figura tridimensional
 */
public class Cubo extends Figura3d {
    
    private double lado;
    
    /**
     * Constructor de la clase Cubo
     * @param lado Longitud del lado del cubo
     * @param idFigura Identificador único de la figura
     */
    public Cubo(double lado, int idFigura) {
        super("Cubo", idFigura);
        this.lado = lado;
    }
    
    /**
     * Calcula el volumen del cubo
     * @return El volumen del cubo
     */
    @Override
    public double calcularVolumen() {
        return this.lado * this.lado * this.lado;
    }
    
    /**
     * Calcula el área de superficie del cubo
     * @return El área de superficie del cubo
     */
    public double calcularAreaSuperficie() {
        return 6 * (this.lado * this.lado);
    }
    
    /**
     * Obtiene la longitud del lado del cubo
     * @return La longitud del lado
     */
    public double getLado() {
        return this.lado;
    }
    
    /**
     * Establece la longitud del lado del cubo
     * @param lado La nueva longitud del lado
     * @throws IllegalArgumentException Si el lado es menor o igual a cero
     */
    public void setLado(double lado) {
        if (lado <= 0) {
            throw new IllegalArgumentException("El lado debe ser mayor que cero");
        }
        this.lado = lado;
    }
    
    /**
     * Representación en cadena del cubo
     * @return Representación del cubo
     */
    @Override
    public String toString() {
        return String.format("Cubo(id=%d, lado=%.2f)", getId(), lado);
    }
}

File: python/Figuras/cubo.py
"""
Clase Cubo - Figura tridimensional
"""
# pylint: disable=invalid-name
from Figura3d import Figura3d

class Cubo(Figura3d):
    """Clase que representa un cubo"""

    def __init__(self, lado: float, id_figura: int):
        """
        Constructor de la clase Cubo

        Args:
            lado (float): Longitud del lado del cubo
            id_figura (int): Identificador único de la figura
        """
        super().__init__("Cubo", id_figura)
        self._lado = lado

    def calcular_volumen(self) -> float:
        """Calcula el volumen del cubo"""
        return self._lado ** 3

    def calcular_area_superficie(self) -> float:
        """Calcula el área de superficie del cubo"""
        return 6 * (self._lado ** 2)

    def get_lado(self) -> float:
        """Obtiene la longitud del lado del cubo"""
        return self._lado

    def set_lado(self, lado: float) -> None:
        """Establece la longitud del lado del cubo"""
        if lado <= 0:
            raise ValueError("El lado debe ser mayor que cero")
        self._lado = lado

    def __str__(self) -> str:
        """Representación en cadena del cubo"""
        return f"Cubo(id={self.get_id()}, lado={self._lado:.2f})"

    def __repr__(self) -> str:
        """Representación técnica del cubo"""
        return f"Cubo(lado={self._lado}, id_figura={self.get_id()})"
